app/controllers/accept_account_invitations_controller.rb

```diff
@@ -1,18 +1,24 @@
 # frozen_string_literal: true
 
 class AcceptAccountInvitationsController < ApplicationController
-  before_action :authenticate_user!
+  def show
+    return if user_signed_in? && the_invitation
 
-  def show; end
+    session[:token] = ps[:token]
+    redirect_to new_user_session_url(email: the_invitation.email)
+  end
 
   def update
     received_invitation.update!(accepted_at: Time.current)
-    redirect_to account_path(received_invitation.account_id)
   end
 
   private
 
   helper_method memoize def received_invitation
     AccountInvitation.unaccepted.for(current_user).find_by!(token: ps.fetch(:token))
   end
+
+  memoize def the_invitation
```

Comment:
Remove this check, if they sign in and the after-sign-in link is wrong, they will see 404 page and this is fine. In other words, the code already handles this scenario and no need to write special code for that.